src/test/java/uk/gov/hmcts/reform/waworkflowapi/duedate/DueDateServiceTest.java

```diff
@@ -0,0 +1,161 @@
+package uk.gov.hmcts.reform.waworkflowapi.duedate;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.waworkflowapi.duedate.holidaydates.HolidayService;
+import uk.gov.hmcts.reform.waworkflowapi.external.taskservice.TaskToCreate;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.reform.waworkflowapi.external.taskservice.Task.PROCESS_APPLICATION;
+
+@SuppressWarnings("PMD.JUnitAssertionsShouldIncludeMessage")
+class DueDateServiceTest {
+
+    public static final String TCW_GROUP = "TCW";
+    private FixedDateService dateService;
+    private DueDateService underTest;
+    private HolidayService holidayService;
+
+    @BeforeEach
+    void setUp() {
+        dateService = new FixedDateService();
+        holidayService = mock(HolidayService.class);
+        underTest = new DueDateService(dateService, holidayService);
+    }
+
+    @Test
+    void haveToSetEitherADueDateOrHaveWorkingDays() {
+        TaskToCreate taskToCreate = new TaskToCreate(PROCESS_APPLICATION, TCW_GROUP);
+        assertThrows(IllegalStateException.class, () -> {
+            underTest.calculateDueDate(
+                null,
+                taskToCreate
+            );
+        });
+
+    }
+
+    @Test
+    void ifADueDateIsAlreadySetDoNotCalculateANewOne() {
+        ZonedDateTime providedDueDate = ZonedDateTime.now();
+        ZonedDateTime calculatedDueDate = underTest.calculateDueDate(
+            providedDueDate,
+            new TaskToCreate(PROCESS_APPLICATION, TCW_GROUP)
+        );
+
+        assertThat(calculatedDueDate, is(providedDueDate));
+    }
+
+    @Test
+    void calculateDueDateAllWorkingDays() {
+        int leadTimeDays = 2;
+        ZonedDateTime now = ZonedDateTime.of(2020, 9, 1, 1, 2, 3, 4, ZoneId.systemDefault());
+        dateService.setCurrentDateTime(now);
+
+        ZonedDateTime calculatedDueDate = underTest.calculateDueDate(
+            null,
+            new TaskToCreate(PROCESS_APPLICATION, TCW_GROUP, leadTimeDays)
+        );
+
+        assertThat(calculatedDueDate, is(now.plusDays(leadTimeDays)));
+    }
+
+    @Test
```

Comment:
maybe we could parametrise these tests and reuse the code. It looks like you repeat the very similar code for each scenario.

Reply:
I've removed a load of duplication from this test. If we were to use a parameterised tests can we still give each row a name which is printed out as part of the test. I think the names of the methods are important here e.g. calculateDueDateWhenStraddlesAWeekend and wonder if there is a way to keep that with a parameterised test?

Reply:
yes, it looks better now. I think it's possible,  I used to do with another parametrise library and don't know with junit 5.  maybe for the next time around:)